### PR TITLE
blockstore: add option to deduplicate by CID

### PR DESCRIPTION
**Pull request by mvdan**
(see commit message)

### CODE REVIEW

**Comment by mvdan** [comment]
FYI @aarshkshah1992 I verified that https://github.com/ipld/go-car/pull/125 passes with this option. Thanks again for providing reproducible tests.

File: v2/blockstore/readwrite.go [hunk 1]
```diff
@@ -52,6 +54,15 @@ func WithIndexPadding(p uint64) Option {
 	}
 }
 
+// WithCidDeduplication makes Put calls ignore blocks if the blockstore already
+// has the exact same CID.
+// This can help avoid redundancy in a CARv1's list of CID-Block pairs.
+//
+// Note that this compares whole CIDs, not just multihashes.
+func WithCidDeduplication(b *ReadWrite) {
```
Comment (aarshkshah1992) [inline v2/blockstore/readwrite.go:62]: @mvdan This is not the default ?

Comment (mvdan) [reply v2/blockstore/readwrite.go:62]: It's still unclear what the default semantics of this blockstore will be. [This thread](https://protocollabs.slack.com/archives/C01MFLTLKTP/p1625142319426200?thread_ts=1624963667.404300&cid=C01MFLTLKTP) goes into some detail.

For now, to unblock us all, I'm keeping this opt-in, and keeping the default behavior relatively close to how go-ipfs implements its blockstores.

Comment (mvdan) [reply v2/blockstore/readwrite.go:62]: I've raised https://github.com/ipld/go-car/issues/130 as a reminder before the release.

**Review by dirkmc — APPROVED** [review]
LGTM 👍

**Pull request merged**